### PR TITLE
feature/FRON-1473: Payment instructions is sent to the customer's ema…

### PR DESCRIPTION
**Pull request by aashishgurung**
#### 1. Objective

Send the payment instructions to the customer's email for convenience store payment method.

Jira Ticket: ([#1473](https://opn-ooo.atlassian.net/browse/FRON-1473))

#### 2. Description of change

Added an email template `convenience_store_template.html` for the convenience store. In `OmiseHelper.php`, the previous `offlinePaymentMethods` property is changed to `imageCodePaymentMethods` so that `offlinePaymentMethods` could include Convenience store as well. Method `isPayableByImageCode` is replaced by `offlinePaymentMethods` in the `PaymentCreationObserver` since we have to send email for convenience store as well.

#### 3. Quality assurance

Make sure your store is setup for Japan. Place an order using convenience store. You should see following email
Screen Shot 2565-05-26 at 16 40 10

**🔧 Environments:**

- Platform version: Magento CE 2.4.3
- Omise plugin version: Omise-Magento 2.23.2
- PHP version: 7.1.15

### CODE REVIEW

**Comment by aashishgurung** [comment]
Some changes conflicts with the change done in [PR #351](https://github.com/omise/omise-magento/pull/351). Need to include changes those changes once the PR #351 is merged.

**Review by kan98 — APPROVED** [review]

File: view/frontend/email/convenience_store_template.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<!--@subject {{trans "Convenience Store Payment"}}  @-->
+{{template config_path="design/email/header_template"}}
+<div>
+    <p>
+        {{trans "%thankyou" thankyou="Thank you for your order."}}</p>
+        {{trans "%message" message="You are now one step away from completing the order. Please follow the link to find the payment instructions to complete the payment "}}.
```
Comment (som-m) [inline view/frontend/email/convenience_store_template.html:6]:
```suggestion
        {{trans "%message" message="You are now one step away from completing the order. Please follow the link to find the payment instructions to complete the payment."}}
```

File: view/frontend/email/convenience_store_template.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    </p>
+    <p>
+        <br /><br /><a style="color: black; border-style: solid; border-width: 1px 1px 1px 1px; text-decoration: none; padding: 4px; border-radius: 5px" href="{{var data.link|raw }}" _blank="">
+            {{trans "%paymentInstructions" paymentInstructions="Payment Instructions."}}
```
Comment (som-m) [inline view/frontend/email/convenience_store_template.html:10]:
```suggestion
            {{trans "%paymentInstructions" paymentInstructions="Payment Instructions"}}
```

**Review by tookaa2 — APPROVED** [review]

File: Helper/OmiseHelper.php [hunk 3]
```diff
@@ -186,23 +196,34 @@ public function convertTescoSVGCodeToHTML($svg)
     }
 
     /**
-     * This method checks and return TRUE if $paymentMethod is offline payment which is payable by image code
-     * otherwise returns false.
+     * Check if payment method is among the payment methods which is payable by image code and
```
Comment (hpariwat) [inline Helper/OmiseHelper.php:199]:
```suggestion
     * Check if payment method is among the payment methods which is payable by image code or not
```

File: view/frontend/email/convenience_store_template.html (hunk 2 above)
Comment (aashishgurung) [reply view/frontend/email/convenience_store_template.html]: @som-m Thanks

File: Helper/OmiseHelper.php (hunk 3 above)
Comment (aashishgurung) [reply Helper/OmiseHelper.php]: @hpariwat Thanks

**Pull request merged**